### PR TITLE
Added support of multiple special groups in regexp

### PR DESCRIPTION
**Pull request by drey**
Hi Alexandr, you made a really useful plugin. I have tried to do it a little bit better.

Now it's possible to use special group (!...) several times. For example, you can search by regular expression like that: "(!ip\sdhcp\spool.+?)\n(?:.+?\n)+?(!\sdns-server.+$)", to get result showed below:

screen shot 2016-09-10 at 15 03 53

Thanks!

### CODE REVIEW

**Comment by mvoidex** [comment]
Nice feature, thanks!
I'll review it (all tabs were replaced with spaces, so diff is useless now), update readme and merge.

**Comment by drey** [comment]
I implemented a feature to select by regexp group without special group notation. I prefer to use classical regexp syntax to work with groups. And of course you can exclude a group from selection by embedded regexp method `(?:..)`. You can merge it if you like.

ps You can see diff between our latest commits by this [link](https://github.com/mvoidex/SublimeSelectByRegex/compare/master...Drey:feature-multiple-groups).

**Comment by mvoidex** [comment]
Merged, thanks for PR!

File: selectbyregex.py [hunk 1]
```diff
@@ -2,53 +2,44 @@
...
-			m = self.rx.search(self.text, min(r.a, r.b))
-			if m and r.a == r.b or m.end() <= max(r.a, r.b) or has_sel(rx): # Restrict to selection if it's not empty and doesn't have $_
-				if self.in_group:
-					if 'Select' in m.groupdict():
-						self.inner_regions.append(sublime.Region(m.start('Select'), m.end('Select')))
-						self.outer_regions.append(sublime.Region(m.start(), m.end()))
+			self.rx = re.compile(unwrap_sel(rx, self.view.substr(r)), re.MULTILINE)
+			m = self.rx.search(self.text, r.a)
+			if m and r.a == r.b or m.end() <= r.b:  # Restrict to selection if it's not empty
+				if len(m.groups()) > 1:
```
Comment (mvoidex) [inline selectbyregex.py:41]: Maybe `len(m.groups()) > 0`?

Comment (drey) [reply selectbyregex.py:41]: Yes, you are right! And if to be honest may be `len(m.groups())` will be clearer.

Thanks!

**Pull request merged**